Source/RapyutaSimulationPlugins/Private/Core/RRStaticMeshComponent.cpp

```diff
@@ -203,59 +208,111 @@ UStaticMesh* URRStaticMeshComponent::CreateMeshBody(const FRRMeshData& InMeshDat
+        check(currentPlatform);
+        const FStaticMeshLODGroup& lodGroup = currentPlatform->GetStaticMeshLODSettings().GetLODGroup(NAME_None);
+        int32 lodsNum = lodGroup.GetDefaultNumLODs();
+        while (staticMesh->GetNumSourceModels() < lodsNum)
         {
-            FMeshSectionInfo info = sectionInfoMap.Get(lodIndex, meshSectionIndex);
-            info.MaterialIndex = 0;
-            sectionInfoMap.Remove(lodIndex, meshSectionIndex);
-            sectionInfoMap.Set(lodIndex, meshSectionIndex, info);
+            staticMesh->AddSourceModel();
         }
-    }
-    staticMesh->SetLightMapResolution(lodGroup.GetDefaultLightMapResolution());
+        for (auto lodIndex = 0; lodIndex < lodsNum; ++lodIndex)
+        {
+            auto& sourceModel = staticMesh->GetSourceModel(lodIndex);
+            sourceModel.ReductionSettings = lodGroup.GetDefaultSettings(lodIndex);
+            sourceModel.BuildSettings.bGenerateLightmapUVs = true;
+            sourceModel.BuildSettings.SrcLightmapIndex = 0;
+            sourceModel.BuildSettings.DstLightmapIndex = 0;
+            sourceModel.BuildSettings.bRecomputeNormals = false;
+            sourceModel.BuildSettings.bRecomputeTangents = false;
+
+            // LOD Section
+            auto& sectionInfoMap = staticMesh->GetSectionInfoMap();
+            for (auto meshSectionIndex = 0; meshSectionIndex < InMeshData.Nodes.Num(); ++meshSectionIndex)
+            {
+                FMeshSectionInfo info = sectionInfoMap.Get(lodIndex, meshSectionIndex);
+                info.MaterialIndex = 0;
+                sectionInfoMap.Remove(lodIndex, meshSectionIndex);
+                sectionInfoMap.Set(lodIndex, meshSectionIndex, info);
+            }
+        }
+        staticMesh->SetLightMapResolution(lodGroup.GetDefaultLightMapResolution());
 #endif
 
-    // Mesh's static materials
-    for (const auto& materialInstance : InMeshData.MaterialInstances)
-    {
-        staticMesh->AddMaterial(materialInstance->GetBaseMaterial());
+        // Mesh's static materials
+        for (const auto& materialInstance : InMeshData.MaterialInstances)
+        {
+            staticMesh->AddMaterial(materialInstance->GetBaseMaterial());
+        }
     }
+
+    // Build mesh
     staticMesh->BuildFromMeshDescriptions({&meshDesc}, meshDescParams);
-    // staticMesh->PostLoad();
+    return staticMesh;
+}
+
+UStaticMesh* URRStaticMeshComponent::CreateMeshBody(const FRRMeshData& InMeshData)
+{
+    UStaticMesh* visualMesh = CreateMesh(InMeshData, true);
+    // Generate complex in case of no simple collision
+    if (false == bUseDefaultSimpleCollision)
+    {
+        auto* bodySetup = visualMesh->GetBodySetup();
+        if (bUseComplexCollision)
+        {
+            visualMesh->ComplexCollisionMesh = CreateMesh(InMeshData, false);
+            bodySetup->CollisionTraceFlag = ECollisionTraceFlag::CTF_UseComplexAsSimple;
+        }
+        else
```

Comment:
Is this elase{} non default simple collision? When we use this?

Reply:
Yes all of those are wrapped inside `if(!bUseDefaultSimpleCollision)`
Basically, Complex collision, as using an external mesh, only works for static objects in UE since moveable one is expensive to simulate
https://docs.unrealengine.com/5.1/en-US/simple-versus-complex-collision-in-unreal-engine
Custom simple collision, as generated by VHACD, works for both static/movable objects.